doc/visual_testing/reporter_visual_inspection_suite.py

```diff
@@ -320,20 +322,74 @@ def report_slm_oasis(build_type):
     return report
 
 
-def report_surface_glm(build_type):
-    """Empyt reports."""
+def report_surface_flm(build_type):
+    """FirstLevelGLM surface reports."""
+    if build_type == "partial":
+        _generate_dummy_html(filenames=["flm_surf_empty.html"])
+        _generate_dummy_html(filenames=["flm_surf.html"])
+        return None
+
     flm = FirstLevelModel(mask_img=SurfaceMasker())
     report_flm_empty = flm.generate_report(height_control=None)
     report_flm_empty.save_as_html(REPORTS_DIR / "flm_surf_empty.html")
 
+    data = fetch_localizer_first_level()
+
+    fsaverage5 = load_fsaverage()
+    surface_image = SurfaceImage.from_volume(
+        mesh=fsaverage5["pial"],
+        volume_img=data.epi_img,
+    )
+
+    t_r = 2.4
+    slice_time_ref = 0.5
+
+    glm = FirstLevelModel(
+        t_r=t_r,
+        slice_time_ref=slice_time_ref,
+        hrf_model="glover + derivative",
+        minimize_memory=False,
+        verbose=1,
+    ).fit(run_imgs=surface_image, events=data.events)
+
+    design_matrix = glm.design_matrices_[0]
+    contrast_matrix = np.eye(design_matrix.shape[1])
+
+    basic_contrasts = {
+        column: contrast_matrix[i]
+        for i, column in enumerate(design_matrix.columns)
+    }
+
+    contrasts = {
+        "(left - right) button press": (
+            basic_contrasts["audio_left_hand_button_press"]
+            - basic_contrasts["audio_right_hand_button_press"]
+            + basic_contrasts["visual_left_hand_button_press"]
+            - basic_contrasts["visual_right_hand_button_press"]
+        )
+    }
+
+    report_flm = glm.generate_report(
+        contrasts,
+        threshold=3.0,
+        bg_img=load_fsaverage_data(data_type="sulcal", mesh_type="inflated"),
+        height_control=None,
+    )
+
+    report_flm.save_as_html(REPORTS_DIR / "flm_surf.html")
+
+    return report_flm, report_flm_empty
+
+
+def report_surface_slm(build_type):
     flm = SecondLevelModel(mask_img=SurfaceMasker())
     report_slm_empty = flm.generate_report(height_control="bonferroni")
     report_slm_empty.save_as_html(REPORTS_DIR / "slm_surf_empty.html")
 
     if build_type == "partial":
         _generate_dummy_html(filenames=["flm_surf_empty.html"])
         _generate_dummy_html(filenames=["slm_surf_empty.html"])
-        return report_flm_empty, report_slm_empty
+        return report_slm_empty
```

Comment:
Are we not returning `report_flm_empty` since we already test it above in `report_surface_flm` ?

Reply:
there was some mix up there indeed: fixing it